lib/rucio/core/rule.py

```diff
@@ -1755,7 +1751,7 @@ def get_updated_dids(total_workers, worker_number, limit=100, blocked_dids=[], *
 
 
 @read_session
-def get_rules_beyond_eol(date_check, worker_number, total_workers, *, session: "Session"):
+def get_rules_beyond_eol(date_check: datetime, worker_number: int, total_workers: int, *, session: "Session") -> list[dict[str, Any]]:
```

Comment:
Also the return type here is wrong.

Reply:
Would this return type be correct?

```
                         list[tuple[InternalScope, 
                                         str,
                                         str,
                                         bool,
                                         str,
                                         datetime,
                                         datetime,
                                         InternalAccount]]
```

Reply:
`ReplicationRule.expires_at` can be `None`. Technically, it’s the same for `eol_at`, but the query filters them out; probably it’s fine to keep that one as is.

The checker may complain about `tuple`, but that can be fixed in the SQLAlchemy migration.